inspirehep/modules/records/tasks.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Records tasks."""
+
+from __future__ import absolute_import, division, print_function
+
+from celery import shared_task
+from elasticsearch.helpers import scan
+from flask import current_app
+from six import iteritems
+
+from invenio_db import db
+from invenio_records.api import Record
+from invenio_search import current_search_client as es
+
+from inspirehep.modules.pidstore.utils import (
+    get_endpoint_from_schema,
+    get_index_from_endpoint,
+)
+
+
+@shared_task(ignore_result=True)
+def update_refs(old, new):
+    records = get_records_to_update(old)
+
+    with db.session.begin_nested():
+        for record in records:
+            update_links(record, old, new)
+            record.commit()
+    db.session.commit()
+
+
+def update_links(record, old, new):
+    def _update_links(record, parts, old, new):
+        for i, part in enumerate(parts):
+            if isinstance(record, dict):
+                try:
+                    record = record[part]
+                except KeyError:
+                    return
+            elif isinstance(record, list):
+                for el in record:
+                    _update_links(el, parts[i:], old, new)
+                return
+
+        if record['$ref'] == old:
+            record['$ref'] = new
+
+    endpoint = get_endpoint_from_schema(record['$schema'])
+    whitelist = current_app.config['REF_UPDATER_WHITELIST'][endpoint]
+
+    for path in whitelist:
+        _update_links(record, path.split('.'), old, new)
+
+
+def get_records_to_update(old):
+    def _get_uuids_to_update(old):
+        def _replace_record_with_recid(path):
+            return path.replace('record', 'recid')
+
+        def _ref_to_recid(ref):
+            return int(ref.split('/')[-1])
+
+        whitelists = current_app.config['REF_UPDATER_WHITELIST']
+
+        for endpoint, whitelist in iteritems(whitelists):
```

Comment:
Do you mean that you prefer to see
```python
import six

...

for k, v in six.iteritems(d):
    ...
```
rather than
```python
from six import iteritems

...

for k, v in iteritems(d):
    ...
```
?

Or maybe you didn't spot that this is `iteritems` from `six`, but you suggested me to use it?